crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -488,7 +488,16 @@ async fn execute_ics20_transfer<S: StateWriteExt>(
 
         // strip the prefix from the denom, as we're back on the source chain
         // note: if this is a refund, this is a no-op.
-        let denom = unprefixed_denom.to_base_denom();
+        let denom = if is_refund {
+            unprefixed_denom
+        } else {
+            unprefixed_denom
+                .remove_prefix(&format!("{}/{}/", source_port, source_channel))
+                .expect(
+                    "denom must be prefixed by source_port/source_channel as it was checked in \
```

Comment:
Is this actually correct?

This is the definition of `fn is_prefixed` (after the changes)
```rust
fn is_prefixed(source_port: &PortId, source_channel: &ChannelId, asset: &Denom) -> bool {
    let prefix = format!("{source_port}/{source_channel}/");
    asset.is_prefixed_with(&prefix)
}
```
But here is `Asset::is_prefixed_with`:
```rust
    pub fn is_prefixed_with(&self, prefix: &str) -> bool {
       self.prefix.starts_with(prefix)
    }
```
So this code is checking `Asset::prefix` if it starts with `"{source_port}/{source_channel}"`, but it actually removing removing `{source_port}/{source_channel}` from the concatenation `{Asset::prefix}/{Asset::denom}`.

Reply:
yes this is the correct functionality, for example if you have a token `{source_port}/{source_channel}/base_denom` then `is_prefixed` is true, and if this is source we want to remove `{source_port}/{source_channel}` from the whole denom. this is the same functionality as before, except now we handle the case where a token has multiple IBC hops, and is something like `{source_port}/{source_channel}/other_port/other_channel/base_denom`. previously, the entire prefix `{source_port}/{source_channel}/other_port/other_channel` would have been removed, which is wrong, we only want to remove the  `{source_port}/{source_channel}` prefix.